### ISSUE TITLE
Aligning let bindings on destructuring

### ISSUE THREAD

**Issue by ricardojmendez** [cursiveclojure/cursive#949]
Cursive is not fully aligning the let bindings if one of them is doing destructuring. 

For example, the following:

```clojure
let [[_ clj-data] (get-request "/api/memory")
      item (first clj-data)
      a (first item)]
```

will be aligned as

```clojure
let [[_ clj-data] (get-request "/api/memory")
      item (first clj-data)
      a    (first item)]
```

instead of

```clojure
let [[_ clj-data] (get-request "/api/memory")
      item        (first clj-data)
      a           (first item)]
```

**Comment by cursiveclojure** [comment]
Yes, this is by design. The problem is that the IntelliJ formatter is a constraint-based algorithm, and it has no way to express "align the let-binding values unless the width of the keys is greater than 20", which is what you'd really want. The problem with destructuring is that the forms can be very long, especially map destructuring. So currently Cursive just doesn't try to align the values where the key is a destructuring form. 

I don't think you'd ever want alignment for map destructurings, but an argument could be made for having a flag allowing it for vector destructurings, but that would mean that something like `[item clj-data [first second] :as all]` will push everything way over - there's no way I can put an upper limit on the width.

**Comment by gnl** [comment]
Maybe add an option under Code Style > Clojure > General > "Align let bindings on destructuring"?
Personally I'd prefer that behaviour most of the time and in cases where the destructuring vector gets too long I can always do something like this:

```
(let [x                (something ...)
      [item clj-data
       [first second] 
       :as all]        data
      y                asdf])
```

Maybe not very pretty, but I think still more readable than this:

```
(let [x (something ...)
      [item clj-data [first second] :as all] data
      y asdf])
```

**Comment by gnl** [comment]
And I'd also apply the option to map destructuring, as one can wrap those over multiple lines just as well.

**Comment by danieroux** [comment]
This is biting me as well. It's one of the last things stopping the team from achieving formatter-nirvana.

We are going full tilt with https://tonsky.me/blog/clojurefmt.

Right now, the Emacs `clojure-mode` configuration and Cursive produces the exact same formatting across the code base - except for this.
